code/ecdsap256sha2/Hacl.Impl.ECDSA.P256SHA256.Signature.Agile.fst

```diff
@@ -0,0 +1,319 @@
+module Hacl.Impl.ECDSA.P256SHA256.Signature.Agile
+
+open FStar.HyperStack.All
+open FStar.HyperStack
+module ST = FStar.HyperStack.ST
+
+open Lib.IntTypes
+open Lib.Buffer
+open Lib.ByteSequence
+
+open Spec.ECDSA
+open Spec.P256.Lemmas
+
+open FStar.Mul
+open FStar.Math.Lemmas
+
+open Hacl.Hash.SHA2
+
+open Spec.P256
+open Spec.P256.Lemmas
+open Spec.P256.Definitions
+
+open Spec.ECDSAP256.Definition
+
+open Hacl.Impl.LowLevel
+
+open Hacl.Impl.P256
+
+open Hacl.Impl.ECDSA.MM.Exponent
+open Hacl.Impl.ECDSA.MontgomeryMultiplication
+
+open Hacl.Impl.P256.Signature.Common
+
+module H = Spec.Agile.Hash
+module Def = Spec.Hash.Definitions
+
+open Spec.Hash.Definitions
+open Hacl.Hash.Definitions
+
+#set-options "--z3rlimit 100"
```

Comment:
I would recommend setting fuel and ifuel to zero here, as mentioned in your previous pull request